File: 0118-pascals-triangle/0118-pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        List<IList<int>> ans = new List<IList<int>>();
        for(int i = 1; i <= numRows; i++){
            int[] arr = new int[i];
            Array.Fill(arr,1);
            for(int j = 0; j < arr.Length; j++){
                if(j != 0 && j != arr.Length - 1){
                    arr[j] = ans[i-2][j-1] + ans[i-2][j];
                }
            }
            List<int> tmp = new List<int>(arr);
            ans.Add(tmp);
        }
        return ans;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        for(int i = 1; i <= numRows; i++){
            vector<int> tmp(i,1);
            for(int j = 0; j < tmp.size(); j++){
                if(j != 0 && j != tmp.size() - 1){
                    tmp[j] = ans[i-2][j-1] + ans[i-2][j];
                }
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};